apprunner-maven-plugin/src/main/java/org/projectnessie/quarkus/maven/QuarkusAppStartMojo.java

```diff
@@ -295,9 +295,10 @@ public void execute() throws MojoExecutionException, MojoFailureException {
 
       Properties projectProperties = getProject().getProperties();
       projectProperties.setProperty(httpListenUrlProperty, listenUrl);
-      projectProperties.setProperty(
-          httpListenPortProperty, Integer.toString(URI.create(listenUrl).getPort()));
-
+      String quarkusPort = Integer.toString(URI.create(listenUrl).getPort());
+      projectProperties.setProperty(httpListenPortProperty, quarkusPort);
+      // set the port as the property, to allow clients of the extension to get the port.
+      System.setProperty("quarkus.http.test-port", quarkusPort);
```

Comment:
Why not use `httpListenPortProperty` for the system property name too?

Also, if we're propagating the port to system properties, why not propagate all of the other properties we control here, like `httpListenUrlProperty`? WDYT?

Reply:
That's the only property needed externally so I thought is enough to propagate the one.
I wanted to keep the name `"quarkus.http.test-port"` to be consistent with the `projectProperties` naming.

Reply:
If we're exposing the port in _project_ properties, could we not have the _parent_ build transfer that value to a system property only when needed?